### PR TITLE
PMM-9367 mongo sharded cluster backup

### PR DESCRIPTION
**Pull request by PavelKhripkov**
PMM-9367

Link to the Feature Build: SUBMODULES-3228

If this PR adds or removes or alters one or more API endpoints, please review and add or update the relevant [API documents](https://github.com/percona/pmm/tree/main/docs/api) as well:

- [ ] API Docs updated

If this PR is related to some other PRs in this or other repositories, please provide links to those PRs:

- Links to related pull requests (optional).

### CODE REVIEW

File: managed/services/management/backup/artifacts_service.go [hunk 1]
```diff
@@ -133,7 +133,11 @@ func (s *ArtifactsService) ListPitrTimeranges(
 	}
 
 	if artifact.Mode != models.PITR {
-		return nil, status.Errorf(codes.FailedPrecondition, "Artifact is not a PITR artifact")
+		return nil, status.Errorf(codes.FailedPrecondition, "Artifact is not a PITR artifact.")
+	}
+
+	if artifact.IsShardedCluster {
+		return nil, status.Errorf(codes.FailedPrecondition, "Getting PITR timeranges not supported for sharded cluster artifacts.")
```
Comment (idoqo) [inline managed/services/management/backup/artifacts_service.go:140]:
```suggestion
		return nil, status.Errorf(codes.FailedPrecondition, "Getting PITR timeranges is not supported for sharded cluster artifacts.")
```

File: managed/services/backup/backup_service.go [hunk 2]
```diff
@@ -524,6 +524,12 @@ func (s *Service) checkArtifactModePreconditions(ctx context.Context, artifactID
 		return errors.Wrapf(ErrArtifactNotReady, "artifact %q in status: %q", artifactID, artifact.Status)
 	}
 
+	if artifact.IsShardedCluster {
+		return errors.Wrapf(ErrIncompatibleService,
+			"artifact %q was taken from sharded cluster and cannot be restored from UI; "+
```
Comment (idoqo) [inline managed/services/backup/backup_service.go:529]: more helpful to link to it directly imo https://docs.percona.com/percona-monitoring-and-management/get-started/backup/backup_mongo.html (or something more helpful).

**Review by idoqo — APPROVED** [review]

File: managed/services/backup/backup_service.go [hunk 3]
```diff
@@ -524,6 +524,12 @@ func (s *Service) checkArtifactModePreconditions(ctx context.Context, artifactID
 		return errors.Wrapf(ErrArtifactNotReady, "artifact %q in status: %q", artifactID, artifact.Status)
 	}
 
+	if artifact.IsShardedCluster {
+		return errors.Wrapf(ErrIncompatibleService,
+			"artifact %q was taken from sharded cluster and cannot be restored from UI; for more information check "+
```
[the remark this answers is not in the document]
Comment (atymchuk) [reply managed/services/backup/backup_service.go:529]: Yes, it can be )

**Review by atymchuk — APPROVED** [review]

File: managed/services/management/backup/artifacts_service.go [hunk 4]
```diff
@@ -133,7 +133,11 @@ func (s *ArtifactsService) ListPitrTimeranges(
 	}
 
 	if artifact.Mode != models.PITR {
-		return nil, status.Errorf(codes.FailedPrecondition, "Artifact is not a PITR artifact")
+		return nil, status.Errorf(codes.FailedPrecondition, "Artifact is not a PITR artifact.")
+	}
+
+	if artifact.IsShardedCluster {
+		return nil, status.Errorf(codes.FailedPrecondition, "Getting PITR timeranges is not supported for sharded cluster artifacts.")
```
Comment (artemgavrilov) [inline managed/services/management/backup/artifacts_service.go:140]: But why?

**Review by artemgavrilov — APPROVED** [review]
LGTM

File: managed/services/management/backup/artifacts_service.go (hunk 4 above)
Comment (PavelKhripkov) [reply managed/services/management/backup/artifacts_service.go:140]: The algorithm that parsing oplog files doesn't split the files by shards, therefore prvides wrong data for sharded setups. You can remember, I mentioned that earlier.

**Pull request merged**